shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -214,50 +214,93 @@ export class CookieManagerTests extends AITestClass {
         this.testCase({
             name: "CookieManager: Set as Disabled",
             test: () => {
+                let getCookieCallCount = 0;
+                let setCookieCallCount = 0;
+                let delCookieCallCount = 0;
+                
                 let manager = createCookieMgr({
                     cookieCfg: {
                         enabled: false,
-                        getCookie: () => { throw "Should not be called" },
-                        setCookie: () => { throw "Should not be called" },
-                        delCookie: () => { throw "Should not be called" }
+                        getCookie: () => { 
+                            getCookieCallCount++;
+                            return "";
+                        },
+                        setCookie: () => { 
+                            setCookieCallCount++;
+                        },
+                        delCookie: () => { 
+                            delCookieCallCount++;
+                        }
                     }
                 });
 
                 let newKey = "test." + newId();
                 let newValue = newId();
-                manager.set(newKey, newValue);
-                Assert.equal("", manager.get(newKey));
-
-                manager.del(newKey);
-                Assert.equal("", manager.get(newKey));
+                
+                // With caching enabled, set should return true and value should be cached
+                let setResult = manager.set(newKey, newValue);
+                Assert.equal(true, setResult, "Set should return true when caching");
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value when disabled");
+
+                // Delete should also be cached
+                let delResult = manager.del(newKey);
+                Assert.equal(true, delResult, "Delete should return true when caching");
+                Assert.equal("", manager.get(newKey), "Should return empty string after cached delete");
+                
+                // Verify that actual cookie functions were not called during disabled phase
+                Assert.equal(0, getCookieCallCount, "getCookie should not be called when disabled");
+                Assert.equal(0, setCookieCallCount, "setCookie should not be called when disabled");
+                Assert.equal(0, delCookieCallCount, "delCookie should not be called when disabled");
             }
         });
 
         this.testCase({
             name: "CookieManager: disable cookies using the legacy setting",
             test: () => {
+                let getCookieCallCount = 0;
```

Comment:
Any tests which is only expecting 0 number of calls and never enables the cookie manager should continue to use the previous pattern -- revert this change in this test